src/Helper/Typo3NodeResolver.php

```diff
@@ -25,36 +24,93 @@ final class Typo3NodeResolver
 
     public function isMethodCallOnGlobals(Node $node, string $methodCall, string $global): bool
     {
-        return $node instanceof MethodCall &&
-               $node->var instanceof ArrayDimFetch &&
-               $this->isName($node->name, $methodCall) &&
-               $this->isName($node->var->var, 'GLOBALS') &&
-               $this->isValue($node->var->dim, $global);
+        if (!$node instanceof MethodCall) {
+            return false;
+        }
+
+        if (!$node->var instanceof ArrayDimFetch) {
+            return false;
+        }
+
+        if (!$this->isName($node->name, $methodCall)) {
+            return false;
+        }
+
+        if (!$this->isName($node->var->var, 'GLOBALS')) {
+            return false;
+        }
+
+        if (null === $node->var->dim) {
+            return false;
+        }
+
+        return $this->isValue($node->var->dim, $global);
```

Comment:
Early return statements make it easier to help static analysis happy and is also much easier to read.

More about Cognitive Complexity: https://www.tomasvotruba.com/blog/2018/05/21/is-your-code-readable-by-humans-cognitive-complexity-tells-you/